site/components/Navigation/NavItemMobile.tsx

```diff
@@ -1,26 +1,48 @@
-import Link from "next/link";
 import React, { FC } from "react";
 import * as styles from "./styles";
-
+import { cx } from "@emotion/css";
+import { ButtonPrimary } from "@klimadao/lib/components";
+import { NavItemMobileID } from "./index";
 interface Props extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
   name: string;
-  url: string;
-  number?: string;
+  toggledNavItemID?: string | undefined;
+  id?: NavItemMobileID;
+  url?: string;
+  setToggledNavItemID?: (string: NavItemMobileID | undefined) => void;
+  subMenu?: JSX.Element[];
+  active?: boolean;
 }
 
 export const NavItemMobile: FC<Props> = (props) => {
+  const activeItem = () => {
+    if (!props.id || !props.toggledNavItemID) {
+      return false;
+    } else {
+      return props.id === props.toggledNavItemID;
+    }
+  };
```

Comment:
No need for a function here IMO.

Here is my suggestion for this Component.

```suggestion

export const NavItemMobile: FC<Props> = (props) => {
  const isToggled =
    !!props.id &&
    !!props.toggledNavItemID &&
    props.id === props.toggledNavItemID;

  const onToggleSubmenu = () => {
    if (isToggled) return;

    props.setToggledNavItemID &&
      props.id &&
      props.setToggledNavItemID(props.id);
  };

  return (
    <div className={cx(styles.navMain_MobileItem)}>
      <ButtonPrimary
        href={props.url} // no need to check for undefined if url is undefined
        className={cx(styles.navMain_MobileLink, {
          active: isToggled,
        })}
        label={props.name}
        onClick={onToggleSubmenu}
      />

      {props.subMenu && (
        <div
          className={styles.navMain_MobileExpanded}
          data-show={isToggled.toString()}
        >
          {props.subMenu}
        </div>
      )}
    </div>
  );
};
```

Reply:
Sweet. I did everything except for  if (isToggled) return ; now set togle id to undefined so that the menu closes. Good suggestions, thanks!